sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java

```diff
@@ -235,7 +235,9 @@ public void registerCoderForType(TypeDescriptor<?> type, Coder<?> coder) {
    *
    * @throws CannotProvideCoderException if a {@link Coder} cannot be provided
    */
+  // CHECKSTYLE.OFF: MissingDeprecated
   @Deprecated
+  // CHECKSTYLE.ON: MissingDeprecated
```

Comment:
Maybe if all the uses are just special cases that don't require real coder inference, that could be fine. But this method itself is quite simple so probably specialized versions would add complexity. I'm open to whatever, though.